permits/views.py

```diff
@@ -803,6 +810,152 @@ def get(self, request, path, *args, **kwargs):
         return services.download_file(path)
 
 
+@method_decorator(login_required, name="dispatch")
+@method_decorator(check_mandatory_2FA, name="dispatch")
+class ArchivedPermitRequestListView(SingleTableMixin, ListView):
+    model = models.ArchivedPermitRequest
+    template_name = "permits/archived_permit_request_list.html"
+
+    error_message = _("Vous n'avez pas les permissions pour archiver cette demande")
+
+    def post(self, request, *args, **kwargs):
+        action = request.POST.get("action")
+
+        if action == "archive-requests":
+            return self.archive()
+
+    def get_queryset(self):
+        return services.get_archived_request_list_for_user(self.request.user).order_by(
+            "-archived_date"
+        )
+
+    def get_table_class(self):
+        return tables.ArchivedPermitRequestsTable
+
+    def archive(self):
+        permit_request_ids = self.request.POST.getlist("to_archive[]")
+
+        if not permit_request_ids:
+            return HttpResponseNotFound(_("Rien à archiver"))
+
+        department = models.PermitDepartment.objects.filter(
+            group__in=self.request.user.groups.all()
+        ).first()
+
+        if not department.is_backoffice and not department.is_integrator_admin:
+            return JsonResponse(
+                data={"error": True, "message": self.error_message}, status=403
+            )
+
+        with transaction.atomic():
+            for permit_request_id in permit_request_ids:
+                permit_request = services.get_permit_request_for_user_or_404(
+                    self.request.user, permit_request_id
+                )
+                permit_request.archive(self.request.user)
+
+        return JsonResponse({"message": _("Demandes archivées avec succès")})
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(check_mandatory_2FA, name="dispatch")
+class ArchivedPermitRequestDeleteView(DeleteView):
+    model = models.ArchivedPermitRequest
+    success_url = reverse_lazy("permits:archived_permit_request_list")
+
+    success_message = _("L'archive #%s a été supprimé avec succès")
+    error_message = _("Vous n'avez pas les permissions pour supprimer cette archive")
+
+    def post(self, request, *args, **kwargs):
+        archive = models.ArchivedPermitRequest.objects.filter(
+            permit_request=kwargs.get("pk")
+        ).first()
+
+        if not archive:
+            raise SuspiciousOperation
+
+        if not self.request.user == archive.archivist:
```

Comment:
Allow superusers to delete archives ? How about other users that have the correct permissions?

Reply:
I checked with Thierry and Olivier, and for the time being only the owner and superusers can delete archives